src/utils.zig

```diff
@@ -1935,3 +1935,58 @@ fn CastFunctionReturnToAnyopaqueType(Fn: type) type {
 pub fn castFunctionReturnToAnyopaque(function: anytype) *const CastFunctionReturnToAnyopaqueType(@TypeOf(function)) {
 	return @ptrCast(&function);
 }
+
+// MARK: Callback
+pub fn NamedCallbacks(comptime Child: type, comptime Function: type) type {
+	return struct {
+		const Self = @This();
+
+		hashMap: std.StringHashMap(*const Function) = undefined,
+
+		pub fn init() Self {
+			var self = Self{.hashMap = .init(main.globalAllocator.allocator)};
+			inline for(@typeInfo(Child).@"struct".decls) |declaration| {
+				if(@TypeOf(@field(Child, declaration.name)) == Function) {
+					std.log.debug("Registered Callback '{s}'", .{declaration.name});
+					self.hashMap.putNoClobber(declaration.name, &@field(Child, declaration.name)) catch unreachable;
+				}
+			}
+			return self;
+		}
+
+		pub fn deinit(self: *Self) void {
+			self.hashMap.deinit();
+		}
+
+		pub fn getFunctionPointer(self: *Self, id: []const u8) ?*const Function {
+			std.debug.assert(id.len > 0);
+			const pointer = self.hashMap.getPtr(id) orelse {
+				return null;
+			};
+			return pointer.*;
+		}
+	};
+}
+
+test "Callback registers testFunction and expects errors" {
+	const TestFunction = fn(_: i32) void;
+	const TestFunctions = struct {
+		// Callback should register this
+		pub fn testFunction(_: i32) void {}
+	};
+	var testFunctions: NamedCallbacks(TestFunctions, TestFunction) = undefined;
+
+	testFunctions = .init();
+	defer testFunctions.deinit();
+
+	try std.testing.expect(testFunctions.hashMap.count() == 1);
+
+	const fnPtr = testFunctions.getFunctionPointer("testFunction");
+	try std.testing.expect(fnPtr != null);
+	if(fnPtr) |callback| {
+		try std.testing.expect(@TypeOf(callback) == *const TestFunction);
+	}
```

Comment:
You already checked if it's null above.
Even without that you don't a null check when using `@TypeOf()`, which is evaluated at compile time.

And instead of comparing the types, you should compare the value with `&TestFunction.testFunction`, because that's what we are interested in here.